api/helpers/uploadedFileErrorChecks.js

```diff
@@ -37,49 +44,95 @@ function initialiseClamScan(req) {
     } = req._sails.config.upload;
     const clamAvOptions = {
         debugMode: JSON.parse(clamavDebugEnabled) || false,
-        clamscan: {active: false},
+        clamscan: { active: false },
         clamdscan: {
             active: false,
             host: clamavHost,
             port: clamavPort,
         },
     };
-
-    return new NodeClam().init(clamAvOptions);
+    try {
+        return new NodeClam().init(clamAvOptions);
+    } catch (err) {
+        throw new Error();
+    }
 }
 
-async function virusScanAndCheckFiletype(req) {
+async function checkFileType(req) {
     try {
-        if (req.files.length === 0) {
-            req.session.eApp.uploadMessages.noFileUploadedError = true;
-            throw new Error('No files were uploaded.');
+        sails.log.info('Checking file type...');
+        for (const file of req.files) {
+            inDevEnvironment
+                ? await checkLocalFileType(file, req)
+                : await checkS3FileType(file, req);
```

Comment:
This is a conversation we had a while back and the reason I brought up using the document-upload tool, or using that node temp folder tool. 

Because multiple large files can be uploaded and we didn't want to store them in memory, Everything gets uploaded to S3 and if they're good, they get a CLEAN tag. If they're not good, they get deleted or automatically deleted after a while because they don't have a CLEAN tag.

I'm happy to create another ticket to look into this personally I prefer the node temp directory option because, no Java. But if we decide it's beneficial for future projects to make changes to the document upload tool then I'll be happy to do it there instead.

Reply:
the main benefit of using it is that most of it is already written, you just need to add some additional settings/code to it so it works for your service. But further down the line, another FCDO service may want to use it and just has to fiddle with env vars here and there.

Reply:
Yeah this is a fair point. I've make a ticket to look into doing that (https://consular.atlassian.net/browse/EIA-374) but I don't know how long it will take.